crates/gen/src/com_interface.rs

```diff
@@ -28,21 +30,43 @@ impl ComInterface {
     pub fn from_type_name(name: TypeName) -> Self {
         let mut count = BTreeMap::new();
 
-        let methods = name
-            .def
-            .methods()
-            .map(|method| {
+        let mut bases = Vec::new();
+        let mut next = name.def;
+
+        loop {
+            let base = next.interfaces().next().unwrap().interface().resolve();
+
+            if base.name() == ("Windows.Win32.Com", "IUnknown") {
+                break;
+            }
+
+            bases.insert(0, TypeName::new(&base, Vec::new(), name.namespace));
+            next = bases[0].def;
+        }
+
+        let mut methods = Vec::new();
+
+        for def in bases
+            .iter()
+            .map(|name| name.def)
+            .chain(std::iter::once(name.def))
+        {
+            for method in def.methods() {
                 let count = count.entry(method.name()).or_insert(0);
                 *count += 1;
 
-                Method {
+                methods.push(Method {
```

Comment:
You can get rid of this by just making the for loop a `map` and then using `collect` at the end. This will could be slightly  more efficient as it might pre-allocate enough space on the heap all at once.

Reply:
See above.